src/wasm/wasm-binary.cpp

```diff
@@ -132,6 +141,24 @@ void WasmBinaryWriter::finishSection(int32_t start) {
       }
     }
   }
+
+  if (binaryLocationsSizeAtSectionStart != binaryLocations.size()) {
+    // We added the binary locations, adjust them: they must be relative
+    // to the code section.
+    assert(binaryLocationsSizeAtSectionStart == 0);
+    // The actual section start is right before the LEB for the size; we want
```

Comment:
this means that the "actual" section start is at the start of the size, but not at the section type? Is that common in other tools/elsewehre in Binaryen? It seems really counterintuitive to me. I would think of the section as starting after the "header" which is the type + size. It's confusing that the payload includes the name, but that's a separate issue?

Reply:
That comment is confusing, yeah, I'll reword. Basically we want to have offsets relative to the "body", which is after the section type byte and the size LEB.